packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -298,27 +364,44 @@ export function resetHooks(): void {
 
   if (__DEV__) {
     currentHookNameInDev = undefined;
+    verifyHookCallOrder();
   }
+  currentHookType = null;
+  currentHookMatches = [];
 
   didScheduleRenderPhaseUpdate = false;
   renderPhaseUpdates = null;
   numberOfReRenders = -1;
 }
 
 function createHook(): Hook {
-  return {
-    memoizedState: null,
+  let hook: Hook = __DEV__
+    ? {
```

Comment:
For things like this it's nice to declare two types, `Hook` and `HookDev`. Then when you need a DEV-only field, do `((hook: any): HookDev).something = ...`

Example of this pattern: https://github.com/facebook/react/blob/8c1614a2fdbb841f65ebf403d7ece3a518c6a984/packages/react-dom/src/server/ReactPartialRenderer.js#L697-L699

Reply:
I'll do this.